### PR TITLE
Upgrade code to make compatible with Lagom 1.4

### PR DESCRIPTION
**Pull request by ignasi35**
This applies changes from https://github.com/lagom/online-auction-java/pull/100 which are compatible with Lagom 1.3 so that final upgrade to Lagom 1.4 is as small as possible.

### CODE REVIEW

**Comment by ignasi35** [comment]
I could apply more than I expected: I only left `Validatable` out (validation in Forms).

File: web-gateway/app/controllers/ItemController.java [hunk 1]
```diff
@@ -33,23 +35,30 @@
     private final BiddingService bidService;
 
     private final Boolean showInlineInstruction;
+    private HttpExecutionContext ec;
```
Comment (TimMoore) [inline web-gateway/app/controllers/ItemController.java]: Did you ask anyone on the Play team about the need to specify an explicit `HttpExecutionContext`? It seems like it has created a very ugly API 😦

Comment (ignasi35) [inline web-gateway/app/controllers/ItemController.java]: Yeah, I did. This was added in Play 2.5 but it didn't create any deprecation warning.
There are now separate Execution contexts and you must use the HTTP one for http rendering.

Comment (lakhina) [inline web-gateway/app/controllers/ItemController.java]: Agree on its weird nature

File: web-gateway/app/controllers/SearchController.java [hunk 2]
```diff
@@ -29,34 +31,38 @@
     private final SearchService searchService;
 
     private final Boolean showInlineInstruction;
+    private HttpExecutionContext ec;
 
     @Inject
-    public SearchController(Configuration config,
+    public SearchController(Config config,
```
Comment (lakhina) [inline web-gateway/app/controllers/SearchController.java]: Shouldn't any lib avoid this kind of changes?

**Comment by lakhina** [comment]
I have got to understand what is PR meant to say but not able to understand reason for this complexity

File: web-gateway/app/controllers/SearchController.java (hunk 2 above)
Comment (ignasi35) [inline web-gateway/app/controllers/SearchController.java]: Yes, and they did.

`Configuration` may still be used but is deprecated. In order to reduced the number of deprecation warnings I made the update to the suggested new API.

**Comment by TimMoore** [comment]
@lakhina these changes were all made for compatibility with Play 2.6 (which Lagom 1.4 depends on).

**Comment by ignasi35** [comment]
There's enough `approve`, I'll merge :-P

**Pull request merged**